File: src/main/java/net/eduard/api/lib/plugin/PluginSettings.java
package net.eduard.api.lib.plugin;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class PluginSettings {
    public static long BACKUP_MIN_SECONDS = 60 * 60;
    private boolean debug = true;
    private boolean autoSave = false;
    private long autoSaveSeconds = 60;
    private long autoBackupSeconds = BACKUP_MIN_SECONDS;
    private long lastBackup = System.currentTimeMillis();
    private transient long lastSave = System.currentTimeMillis();

    public long getAutoBackupSeconds() {
        if (autoBackupSeconds < BACKUP_MIN_SECONDS) {
            autoBackupSeconds = BACKUP_MIN_SECONDS;
        }
        return autoBackupSeconds;
    }
    public boolean isAutoSave() {
        return autoSave;
    }
    public long getLastBackup() {
        return lastBackup;
    }

    public boolean isDebug() {
        return debug;
    }

    public void setLastBackup(long lastBackup) {
        this.lastBackup = lastBackup;
    }

    public void setLastSave(long lastSave) {
        this.lastSave = lastSave;
    }

    public long getLastSave() {
        return lastSave;
    }
}

File: src/main/java/net/eduard/api/lib/plugin/PluginSettings.kt
package net.eduard.api.lib.plugin

import net.eduard.api.lib.config.StorageType

class PluginSettings {
    var storeType = StorageType.YAML
    var debug = true
    var autoSave = false
    var autoSaveSeconds = 1
    var autoBackup = false
    var autoBackupSeconds = 1
    @Transient
    var lastBackup = 0L
    @Transient
    var lastSave = 0L
}